File: Backend/utils/chapter_title.py
import re


def format_chapter_number(number) -> str:
    """1.0 → '1', 1.5 → '1.5'."""
    try:
        n = float(number)
    except (TypeError, ValueError):
        return str(number or "").strip()
    if n.is_integer():
        return str(int(n))
    return f"{n:g}"


def default_chapter_title(raw, number) -> str:
    """Boş, sadece sayı veya '#1' ise 'Bölüm n' kaydet (bot ile aynı stil)."""
    n = format_chapter_number(number)
    expected = f"Bölüm {n}"
    text = (raw or "").strip()
    if not text:
        return expected
    if re.fullmatch(r"#?\s*\d+(\.\d+)?", text):
        return expected
    return text


def chapter_display_label(raw, number) -> str:
    """Bildirim ve yorumlarda gösterilecek etiket."""
    n = format_chapter_number(number)
    text = (raw or "").strip()
    if not text:
        return f"Bölüm {n}"
    if re.match(rf"^Bölüm\s+{re.escape(n)}\b", text, flags=re.IGNORECASE):
        return text
    if re.fullmatch(r"#?\s*\d+(\.\d+)?", text):
        return f"Bölüm {n}"
    return f"Bölüm {n} - {text}"

File: frontend/src/lib/chapterTitle.js
export function formatChapterNumber(n) {
  const num = Number(n);
  if (!Number.isFinite(num)) return String(n ?? "").trim();
  if (Number.isInteger(num)) return String(num);
  return String(num);
}

export function isAutoChapterTitle(text) {
  return /^Bölüm\s+[\d.]+$/i.test(String(text || "").trim());
}

export function defaultChapterTitle(raw, number) {
  const n = formatChapterNumber(number);
  const expected = `Bölüm ${n}`;
  const text = String(raw || "").trim();
  if (!text) return expected;
  if (/^#?\s*\d+(\.\d+)?$/.test(text)) return expected;
  return text;
}
